compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -175,8 +203,16 @@ pub mod dep_kind {
                 const is_anon: bool = contains_anon_attr!($($attrs)*);
                 const is_eval_always: bool = contains_eval_always_attr!($($attrs)*);
 
+                #[inline(always)]
+                fn can_reconstruct_query_key() -> bool {
+                    !is_anon &&
+                    <query_keys::$variant<'_> as DepNodeParams<TyCtxt<'_>>>
+                        ::can_reconstruct_query_key()
```

Comment:
I might be wrong, but this looks like it has different behavior to before - I'm not seeing a loop over the tuple_arg_ty here. Is that intentional?

Reply:
The loop over `tuple_arg_ty` was actually a loop over a single element. See rustc_macros/src/query.rs:480.
Directly using `query_keys` seemed simpler.

Reply:
Ok, seems fine then. Thanks!